spec/requests/providers/partners/bank_statements_controller_spec.rb

```diff
@@ -0,0 +1,483 @@
+
+        it "does not add attachment object" do
+          expect { request }.not_to change(legal_aid_application.attachments, :count)
+        end
+
+        it "does not enqueue job to convert upload to pdf" do
+          expect { request }.not_to change(PdfConverterWorker.jobs, :size)
+        end
+
+        it "returns http success" do
+          request
+          expect(response).to have_http_status(:ok)
+        end
+
+        it "displays error indicating file is of the wrong content type" do
+          request
+          expect(response.body).to have_selector("h2", text: "There is a problem").and have_link("zip.zip must be a DOC, DOCX, RTF, ODT, JPG, BMP, PNG, TIF, CSV or PDF")
+        end
+      end
+
+      context "when file contains a virus", clamav: true do
+        let(:file) { uploaded_file("spec/fixtures/files/malware.doc") }
+
+        it "does not add attachment object" do
+          expect { request }.not_to change(legal_aid_application.attachments, :count)
+        end
+
+        it "does not enqueue job to convert upload to pdf" do
+          expect { request }.not_to change(PdfConverterWorker.jobs, :size)
+        end
+
+        it "returns http success" do
+          request
+          expect(response).to have_http_status(:ok)
+        end
+
+        it "displays error indicating file contains a virus" do
+          request
+          expect(response.body).to have_selector("h2", text: "There is a problem").and have_link("malware.doc contains a virus")
+        end
+      end
+
+      context "when virus scanner is down" do
+        let(:file) { uploaded_file("spec/fixtures/files/acceptable.pdf", "application/pdf") }
+
+        before do
+          malware_scan_result = instance_double(MalwareScanResult, virus_found?: false, scanner_working: false)
+          allow(MalwareScanner).to receive(:call).and_return(malware_scan_result)
+        end
+
+        it "returns error message" do
+          request
+          expect(response.body).to include("There was a problem uploading your file - try again")
+        end
+      end
+    end
+
+    context "when save and continue button clicked" do
+      subject(:request) do
+        patch(providers_legal_aid_application_partners_bank_statements_path(legal_aid_application),
+              params: { continue_button: "Save and continue" })
+      end
+
+      context "with files already attached" do
+        before do
+          patch(providers_legal_aid_application_partners_bank_statements_path(legal_aid_application),
+                params: { upload_button: "Upload",
+                          original_file: uploaded_file("spec/fixtures/files/acceptable.pdf", "application/pdf") })
+        end
+
+        it "does not add attachment object" do
+          expect { request }.not_to change(legal_aid_application.attachments, :count)
+        end
+
+        # Commented out as partner HMRC response has not been built yet, but will be added soon
```

Comment:
not a blocker - if this is copied from somewhere else do we need it in here now?